packages/interactivity-router/src/head.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Helper to update only the necessary tags in the head.
+ *
+ * @async
+ * @param {Array} newHead The head elements of the new page.
+ *
+ * @return {void}
+ */
+export const updateHead = async ( newHead ) => {
+	// Helper to get the tag id store in the cache.
+	const getTagId = ( tag ) => tag.id || tag.outerHTML;
+
+	// Map incoming head tags by their content.
+	const newHeadMap = new Map();
+	for ( const child of newHead ) {
+		newHeadMap.set( getTagId( child ), child );
+	}
+
+	const toRemove = [];
+
+	// Detect nodes that should be added or removed.
+	for ( const child of document.head.children ) {
+		const id = getTagId( child );
+		// Always remove styles and links as they might change.
+		if ( child.nodeName === 'LINK' || child.nodeName === 'STYLE' )
+			toRemove.push( child );
+		else if ( newHeadMap.has( id ) ) newHeadMap.delete( id );
+		else if ( child.nodeName !== 'SCRIPT' && child.nodeName !== 'META' )
+			toRemove.push( child );
+	}
+
+	// Prepare new assets.
+	const toAppend = [ ...newHeadMap.values() ];
+
+	// Apply the changes.
+	toRemove.forEach( ( n ) => n.remove() );
+	document.head.append( ...toAppend );
+};
+
+/**
+ * Fetches and processes head assets (stylesheets and scripts) from a specified document.
+ *
+ * @async
+ * @param {Document} document     The document from which to fetch head assets. It should support standard DOM querying methods.
+ * @param {Map}      headElements A map of head elements to modify tracking the URLs of already processed assets to avoid duplicates.
+ *
+ * @return {void}
```

Comment:
It returns an array of HTML elements, for example [`HTMLTitleElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTitleElement) that inherits from `HTMLElement`. That probably should be:

```suggestion
 * @return {HTMLElement[]}
```

Reply:
You're totally right. Copying and pasting issues 😄 I changed it in [this commit](https://github.com/WordPress/gutenberg/pull/59707/commits/743926f0160e985a4641c3b158f64700e8c656ea).